samcli/lib/test_runner/fargate_testsuite_runner.py

```diff
@@ -0,0 +1,468 @@
+            taskDefinition=task_definition_arn
+        )
+        container_definitions = describe_task_definition_response.get("taskDefinition").get("containerDefinitions")
+
+        if not container_definitions:
+            raise InvalidTestRunnerTemplateException(
+                f"The Test Runner Stack {self.runner_stack_name} task definition ({task_definition_arn}) does not contain a container definition."
+            )
+
+        if len(container_definitions) > 1:
+            raise InvalidTestRunnerTemplateException(
+                f"The Test Runner Stack {self.runner_stack_name} task definition ({task_definition_arn}) contains multiple container definitions."
+            )
+
+        container_name = container_definitions[0].get("name")
+        return container_name
+
+    def _get_subnets(self) -> List[str]:
+        """
+        Queries describe-subnets to get subnets associated with the customer account.
+
+        NOTE: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeSubnets.html
+
+        Parameters
+        ----------
+        boto_client_provider : BotoProviderType
+            Provides a boto3 client in order to query the describe-subnets API
+
+        Returns
+        -------
+        List[str]
+            A list of subnet ids associated with a customer account.
+        """
+        subnets = self.boto_ec2_client.describe_subnets().get("Subnets")
+        return [subnet["SubnetId"] for subnet in subnets]
+
+    def _upload_tests_and_reqs(self, bucket: str, temp_tarfile_name: str = str(uuid.uuid4())) -> None:
+        """
+        Compress and upload tests and requirements to an S3 Bucket for the Fargate container to pick up.
+
+        Parameters
+        ----------
+        bucket : str
+            The bucket to which tests and requirements are uploaded.
+
+            This is passed in because it is not known at the time of FargateTestsuiteRunner creation where the bucket name will come from.
+            If a bucket override is not provided, the bucket contained in the Test Runner Stack must be used. The stack must be created/updated before this
+            bucket can be fetched.
+
+        temp_tarfile_name : str
+            The name of the temporary tarball that the tests are compressed into. Once uploaded to the bucket, the tarball is removed.
+
+            Exists as a parameter for testing.
+        """
+        LOG.info(
+            self.color.yellow(
+                f"=> Compressing and uploading {self.tests_path} to {Path(bucket).joinpath(self.path_in_bucket).as_posix()}\n"
+            )
+        )
+
+        # Compress tests into a temporary tarfile to send to S3 bucket
+        # We set arcname to the stem of the tests_path to avoid including all the parent directories in the tarfile
+        # E.g. If the customer specifies tests_path as a/b/c/tests, we want the tar to expand as tests, not a
+        with tarfile.open(temp_tarfile_name, "w:gz") as tar:
```

Comment:
There doesn't seem to be any compression here, and I think that's something we'd want if we plan to upload the tests to a bucket. 
https://github.com/aws/aws-sam-cli/blob/4543732bf3c0da3b57fe1e5aa43ce3f41d2bd0ba/samcli/lib/utils/tar.py#L27
Especially if customer testsuites are decently large, it could make a difference?

Reply:
Why not add an option to do that instead of doing almost the same thing? If we make an improvement or find a bug with how we tar, we then have to update in multiple locations.

Reply:
Sure, will do